src/core/tests/functional/test_config.py

```diff
@@ -272,3 +272,144 @@ def test_delete_report_item_type(self, client, auth_header, cleanup_report_item_
+        response = self.assert_post_ok(client, uri="publishers-presets", json_data=cleanup_publisher_preset, auth_header=auth_header)
+        assert response.json["message"] == "Publisher preset created successfully"
+        assert response.json["id"] == "44"
+
+    def test_modify_publisher_preset(self, client, auth_header, cleanup_publisher_preset):
+        publisher_data = {"description": "new description"}
+        publisher_preset_id = cleanup_publisher_preset["id"]
+        response = self.assert_put_ok(
+            client, uri=f"publishers-presets/{publisher_preset_id}", json_data=publisher_data, auth_header=auth_header
+        )
+        # TODO: again - successful modification returns None (null)
+        assert response.json is None
+
+    def test_get_publisher_preset(self, client, auth_header, cleanup_publisher_preset):
+        response = self.assert_get_ok(client, uri=f"publishers-presets?search={cleanup_publisher_preset['name']}", auth_header=auth_header)
+        assert response.json["items"][0]["name"] == cleanup_publisher_preset["name"]
+        assert response.json["items"][0]["id"] == cleanup_publisher_preset["id"]
+        assert response.json["items"][0]["description"] == "new description"
+        assert response.json["items"][0]["type"] == cleanup_publisher_preset["type"]
+        assert response.json["items"][0]["parameters"]["FTP_URL"] == cleanup_publisher_preset["parameters"]["FTP_URL"]
+
+    def test_delete_publisher_preset(self, client, auth_header, cleanup_publisher_preset):
+        publisher_preset_id = cleanup_publisher_preset["id"]
+        response = self.assert_delete_ok(client, uri=f"publishers-presets/{publisher_preset_id}", auth_header=auth_header)
+        assert response.json["message"] == "PublisherPreset 44 deleted"
+
+
+class TestAttributes(BaseTest):
+    base_uri = "/api/config"
+
+    def test_create_attribute(self, client, auth_header, cleanup_attribute):
+        response = self.assert_post_ok(client, uri="attributes", json_data=cleanup_attribute, auth_header=auth_header)
+        assert response.json["id"] == 42
+        assert response.json["message"] == "Attribute added"
+
+    def test_modify_attribute(self, client, auth_header, cleanup_attribute):
+        attribute_data = {"name": "Attributify McAttributeFace"}
+        attribute_id = cleanup_attribute["id"]
+        response = self.assert_put_ok(client, uri=f"attributes/{attribute_id}", json_data=attribute_data, auth_header=auth_header)
+        # TODO really weird to return None (null)
+        assert response.json is None
+
+    def test_get_attribute(self, client, auth_header, cleanup_attribute):
+        attribute_id = cleanup_attribute["id"]
+        response = self.assert_get_ok(client, uri=f"attributes/{attribute_id}", auth_header=auth_header)
+        print(cleanup_attribute["type"])
+        print(response.json)
+        # TODO: attributes?search=STRING does not return anything.
```

Comment:
replace with "TODO: return value is formatted not correctly - see bug issue "

Reply:
here I have a question: On line 397, is it an expected way to use /attributes/id, or should the request be sent to /attributes?search= --> because that search seems to use the `type` key (of the json). But for /attributes?search=ENUM it returns inconsistently.
Additionally, the entries created by POST, like:
```
{
        "id": 44,
        "name": "TEST Attribute name",
        "description": "Simple attribute desc TEST",
        "type": "STRING"
}
```
are actually unsearchable using the ?search=STRING.
That's actually the reason why I access it using the ID directly. Please check, @ewoiset. thx

Reply:
please add it to https://github.com/ait-cs-IaaS/Taranis-NG/issues/168